guide/src/function.md

```diff
@@ -233,3 +249,41 @@ fn module_with_fn(py: Python, m: &PyModule) -> PyResult<()> {
 
 # fn main() {}
 ```
+
+## Accessing the FFI functions
+
+In order to make Rust functions callable from Python, PyO3 generates a 
+`extern "C" Fn(slf: *mut PyObject, args: *mut PyObject, kwargs: *mut PyObject) -> *mut Pyobject`
+function and embeds the call to the Rust function inside this FFI-wrapper function. This
+wrapper handles extraction of the regular arguments and the keyword arguments from the input
+`PyObjects`. Since this function is not user-defined but required to build a `PyCFunction`, PyO3
+offers the `raw_pycfunction!()` macro to get the identifier of this generated wrapper.
+
+E.g.:
+
+```rust
+use pyo3::prelude::*;
+use pyo3::types::PyCFunction;
+use pyo3::raw_pycfunction;
+
+#[pyfunction]
+fn some_fun() -> PyResult<()> {
+    Ok(())
+}
+
+#[pymodule]
+fn module(_py: Python, module: &PyModule) -> PyResult<()> {
+    let ffi_wrapper_fun = raw_pycfunction!(some_fun);
+    let docs = "Some documentation string with null-termination\0";
+    let py_cfunction =
+        PyCFunction::new_with_keywords(ffi_wrapper_fun, "function_name", docs, module.into())?;
+    module.add_function(py_cfunction)
+}
+
+# fn main() {}
+```
```

Comment:
I think @kngwyu is correct; this document is great to go as a docstring on `raw_pycfunction` or maybe even `PyCFunction::new_with_keywords.

And then in the paragraph above can just have a link e.g. `(See the docs for [raw_pycfunction](https://docs.rs/pyo3/0.12.0/pyo3/macro.raw_pycfunction.html)).

I can't decide whether it's better for `raw_pycfunction` or `PyCFunction::new_with_keywords` though - happy for you to decide!